File: Api/src/Api.Domain/Entities/ChatEntity.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Domain.Entities {
    public class ChatEntity : BaseEntity
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        [MaxLength(7)]
        public string Color { get; set; }

        [JsonIgnore]
        public IEnumerable<MessageEntity> Messages { get; set; }

        [JsonIgnore]
        public IEnumerable<UserChatEntity> UserChats { get; set; }
    }
}

File: App/src/app/chat/entities/chat.entity.ts
import { BaseEntity } from '../../shared/entities/base.entity';
import { UserChatEntity } from '../../shared/entities/userChat.entity';
import { MessageEntity } from './message.entity';

export interface ChatEntity extends BaseEntity {
  name: string;
  color: string;
  messages?: MessageEntity[];
  userChats?: UserChatEntity[];
}
